app/controllers/ContactCheckYourAnswersController.scala

```diff
@@ -49,16 +55,23 @@ class ContactCheckYourAnswersController @Inject() (
   }
 
   def saveAndContinue: Action[AnyContent] = (identify andThen getData andThen requireData) async { implicit request =>
-    val contactInfos = service.getContactInfos(request.userAnswers)
-    request.userAnswers
-      .set(ContactsPage, contactInfos)
+    form
+      .bindFromRequest()
       .fold(
-        error => Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad())),
-        updatedAnswers => {
-          sessionRepository
-            .set(updatedAnswers)
-            .map { _ => Redirect(routes.IndexController.onPageLoad()) }
-        }
+        _ =>
+          Future.failed(
+            BadRequestException("invalid ContactCheckYourAnswersForm submitted")
+          ),
+        dataReceived =>
+          if service.getContactInfos(request.userAnswers) == dataReceived then
+            for {
+              response <- Future
+                .fromTry(request.userAnswers.set(ContactsPage, dataReceived))
+              _ <- sessionRepository.set(request.userAnswers)
```

Comment:
There is a bug here, should be:
```suggestion
              updatedUserAnswers <- Future
                .fromTry(request.userAnswers.set(ContactsPage, dataReceived))
              _ <- sessionRepository.set(updatedUserAnswers)
```
we will also need to update the controller spec to verify what we persist into mongo

Reply:
Fixed in [e6ee71a](https://github.com/hmrc/senior-accounting-officer-registration-frontend/pull/45/commits/e6ee71a587aaca770404cd2b31a25ff633f0588d)